src/viking.js

```diff
@@ -1,11 +1,56 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+      this.health = health;
+      this.strength = strength;
+    }
+       //methods
+  attack(){
+  
+    return this.strength;
+  };
+  receiveDamage(damage){
+      this.health = this.health - damage;
+  };
+  
+}
+  
+  // Viking
+  class Viking extends Soldier {
+    constructor(name, health, strength){
+      super(health, strength)
+      this.name = name;
+    }
+    receiveDamage(damage){
+        this.health = this.health - damage;
+        if(Viking){
```

Comment:
The coerced boolean value of class Viking is assessed here. Is this what you want? Is, perhaps, the health of _this_ of importance?